specification/appconfiguration/AppConfiguration/routes.tsp

```diff
@@ -0,0 +1,973 @@
+https://aka.ms/azconfig/docs/restapirevisions
+""")
+    @query("label")
+    label?: string;
+
+    ...syncTokenHeader;
+
+    @doc("""
+Instructs the server to return elements that appear after the element referred
+to by the specified token.
+""")
+    @query("After")
+    after?: string;
+
+    @doc("""
+Requests the server to respond with the state of the resource at the specified
+time.
+""")
+    @header("Accept-Datetime")
+    acceptDatetime?: string;
+
+    @doc("Used to select what fields are present in the returned resource(s).")
+    @query({
+      name: "$Select",
+      format: "csv",
+    })
+    select?: KeyValueFields[];
+
+    #suppress "@azure-tools/typespec-azure-core/prefer-csv-collection-format" "Pre-existing API contract"
+    @doc("""
+A filter used to query by tags. Syntax reference:
+https://aka.ms/azconfig/docs/restapirevisions
+""")
+    @query({
+      name: "tags",
+      format: "multi",
+    })
+    tags?: string[];
+  },
+  KeyValueListResult &
+    AppConfigResponseHeaders &
+    contentTypeHeader<"application/vnd.microsoft.appconfig.kvset+json">,
+  {},
+  Error
+>;
+
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "No data-plane HEAD operation template"
+@summary("Requests the headers and status of the given resource.")
+@doc("Requests the headers and status of the given resource.")
+@route("/revisions")
+@head
+op checkRevisions is Foundations.Operation<
+  {
+    @doc("""
+A filter used to match keys. Syntax reference:
+https://aka.ms/azconfig/docs/restapirevisions
+""")
+    @query("key")
+    key?: string;
+
+    @doc("""
+A filter used to match labels. Syntax reference:
+https://aka.ms/azconfig/docs/restapirevisions
+""")
+    @query("label")
+    label?: string;
+
+    ...syncTokenHeader;
+
+    @doc("""
+Instructs the server to return elements that appear after the element referred
+to by the specified token.
+""")
+    @query("After")
+    after?: string;
+
+    @doc("""
+Requests the server to respond with the state of the resource at the specified
+time.
+""")
+    @header("Accept-Datetime")
+    acceptDatetime?: string;
+
+    @doc("Used to select what fields are present in the returned resource(s).")
+    @query({
+      name: "$Select",
+      format: "csv",
+    })
+    select?: KeyValueFields[];
+
+    #suppress "@azure-tools/typespec-azure-core/prefer-csv-collection-format" "Pre-existing API contract"
+    @doc("""
+A filter used to query by tags. Syntax reference:
+https://aka.ms/azconfig/docs/restapirevisions
+""")
+    @query({
+      name: "tags",
+      format: "multi",
+    })
+    tags?: string[];
+  },
+  OkResponse & AppConfigResponseHeaders,
+  {},
+  Error
+>;
+
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Pre-existing API contract"
+@summary("Gets the state of a long running operation.")
+@doc("Gets the state of a long running operation.")
+@route("/operations")
+@get
+op getOperationDetails is Foundations.Operation<
```

Comment:
what could be the potential side effects of changing the name now?

Reply:
Main effect is on documentation. The operation will show up with the updated name in your REST API docs, `getOperationDetails` is a bit generic for what this seems to be.

Reply:
Any problem with SDK generation or our existing SDKs?
cc @jimmyca15 to confirm if the name can be changed.

Reply:
There are client customizations available in client.tsp for the sdks to rename this if it doesnt work for them. Plus, since your existing SDKs have a handwritten convenience layer, once folks decide to migrate this they'll still have to plug these operations in under the hood.

Reply:
It should remain GetOperationDetails.